src/libraries/pkg/Microsoft.NETCore.Platforms/runtimeGroups.props

```diff
@@ -130,6 +130,34 @@
       <Versions>11;12;13</Versions>
     </RuntimeGroup>
 
+    <RuntimeGroup Include="solaris">
+      <Parent>unix</Parent>
+      <Architectures>x64</Architectures>
+      <Versions>11</Versions>
+    </RuntimeGroup>
+
+    <RuntimeGroup Include="illumos">
```

Comment:
Will this be a something we leverage to build portable binaries (similar to `linux-x64`)?  If not, should you account for such a RID and should `solaris` import it? cc @janvorli

Reply:
Tested with `dotnet` binary: when compiled on an `illumos` based distro, SmartOS, and copied `artifacts/bin` to other illumos distros: OpenIndiana and OmniOS it worked; but not on Oracle Solaris 11x.

Oracle Solaris from that perspective is a separate platform:

* different syscalls supported by illumos and Solaris kernels.
* they both have their own libc implementations; POSIX compliant but different, so the library loader (ld) looks for different versions and fails if we try to run binary compiled on one, on the other platform. If we satisfy it somehow (local copy of libc and set LD_LIBRARY_PATH to point to its directory), that fails when local libc makes unsupported syscall to kernel.

Both platforms have portable binaries for their own distros/versions, but they do not mix from derivation perspective.

illumos and Oracle Solaris were forked from OpenSolaris in 2010, since then they have diverged.
This is similar to, FreeBSD and OS X used to share same base once, but lately they have diverged.

Reply:
See my comment here https://github.com/dotnet/runtime/pull/37016#discussion_r431413885, for new stuff they are almost certainly no longer binary compatible.

Reply:
Different platform API could hypothetically be shimmed.  I'll defer to @janvorli on the feasibility of doing something like portable-linux here and if we need to think about that from the start.  I suppose it could always be added later.

Reply:
I don't think it makes sense to create a portable thing here - we don't have portability between MUSL and GLIBC based distros for similar reasons.

Reply:
I am referring to illumos vs solaris.

Reply:
Also note that `illumos to solaris` is not only like `glibc to musl` difference (which is two `libc`s for same Linux kernel) but they are also `freebsd to macOS` different (which is two different kernels and their corresponding `libc`s).